src/videojs-media-sources.js

```diff
@@ -724,6 +725,53 @@ deprecateOldCue = function(cue) {
       this.trigger({ type: 'updateend' });
     },
 
+    receiveBuffer_: function(segment) {
+      // create an in-band caption track if one is present in the segment
+      if (segment.captions &&
```

Comment:
This looks like copy-paste of the HTML in-band captions and metadata translation, yes? If so, we should figure out a way to share more of that logic.